### ISSUE TITLE
[false-positive] Can not iterate 'ø' (re‑check type hints)    

### ISSUE THREAD

**Issue by nightlinus** [kalessil/phpinspectionsea#65]
Code example
```php       
$array = array_filter([]);
foreach ($array as $item) {
            
}
```

**Comment by funivan** [comment]
Confirm this bug.

**Comment by kalessil** [comment]
Which PhpStrom/Idea version is it?

**Comment by nightlinus** [comment]
PhpStorm 2016.3.1 EAP
Build #PS-163.9166.11, built on December 1, 2016
JRE: 1.8.0_112-release-408-b2 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

Php Inspections (EA Extended)
Version: 2.2.0

**Comment by kalessil** [comment]
I guess I have to hurry up with fixing this, until it not released...

**Comment by nightlinus** [comment]
I guess this is sarcasm. I am sorry for disturbing, wont open issues for eap versions.
Checked on stable version — no false-positive.

**Comment by kalessil** [comment]
Not at all, I was serious - feedback on EAPs is very valuable to me. Sorry for confusion.

**Comment by kalessil** [comment]
Thank you for reporting @nightlinus , I'm releasing a hotfix version of the plugin today.

**Comment by nightlinus** [comment]
Ty for quick fix, but unfortunetly it is not resolved.
![image](https://cloud.githubusercontent.com/assets/2681887/21040924/ffdaa5cc-bdf8-11e6-9e3e-5aab19a76310.png)

PhpStorm 2016.3.1
Build #PS-163.9735.1, built on December 6, 2016
JRE: 1.8.0_112-release-408-b2 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Php Inspections (EA Extended) Version: 2.2.1

Thank you for your work this is one of the greatest plugins ever, so much value.

**Comment by kalessil** [comment]
@artspb : any hints about the mysterious 'ø' and where it coming from (type resolving)?

**Comment by artspb** [comment]
@neuro159 should be aware of it, please ask him directly.

**Comment by kalessil** [comment]
@artspb : sure, will address all similar topics to Alex  ;)
@neuro159 : great, thank you.

**Comment by b1rdex** [comment]
@neuro159 still present

PhpStorm 2017.1 EAP
Build #PS-171.2152.2, built on January 4, 2017
PhpStorm EAP User
Expiration date: February 3, 2017
JRE: 1.8.0_112-release-b621 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

**Comment by neuro159** [comment]
I know. In my short list...

**Comment by kalessil** [comment]
@b1rdex @neuro159 : is this still present?

PS: I'd then drop my [workaround finally](https://github.com/kalessil/phpinspectionsea/blob/master/src/main/java/com/kalessil/phpStorm/phpInspectionsEA/utils/Types.java#L22) )

**Comment by b1rdex** [comment]
@kalessil got this bug today

PhpStorm 2017.1 EAP
Build #PS-171.3224.10, built on February 15, 2017
PhpStorm EAP User
Expiration date: March 17, 2017
JRE: 1.8.0_112-release-b702 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

![image](https://cloud.githubusercontent.com/assets/312855/23059744/79de890e-f546-11e6-8d61-81368f435ead.png)

**Comment by kalessil** [comment]
Fixed, @b1rdex would you please give a try to plugin version from master?

**Comment by nightlinus** [comment]
Fixed for me with master version. Thank you for fix.

**Comment by kalessil** [comment]
Awesome, thanks for checking!
